packages/next/server/lib/squoosh/codecs.ts

```diff
@@ -259,38 +293,67 @@ export const codecs = {
       max: 100,
     },
   },
+  avif: {
+    name: 'AVIF',
+    extension: 'avif',
+    // eslint-disable-next-line no-control-regex
+    detectors: [/^\x00\x00\x00 ftypavif\x00\x00\x00\x00/],
+    dec: () =>
+      instantiateEmscriptenWasm(avifDec as DecodeModuleFactory, avifDecWasm),
+    enc: async () => {
+      return instantiateEmscriptenWasm(
+        avifEnc as EmscriptenWasm.ModuleFactory<AVIFEncodeModule>,
+        avifEncWasm
+      )
+    },
+    defaultEncoderOptions: {
+      cqLevel: 33,
+      cqAlphaLevel: -1,
+      denoiseLevel: 0,
+      tileColsLog2: 0,
+      tileRowsLog2: 0,
+      speed: 6,
+      subsample: 1,
+      chromaDeltaQ: false,
+      sharpness: 0,
+      tune: 0 /* AVIFTune.auto */,
+    },
+    autoOptimize: {
+      option: 'cqLevel',
+      min: 62,
+      max: 0,
+    },
+  },
   oxipng: {
     name: 'OxiPNG',
     extension: 'png',
     // eslint-disable-next-line no-control-regex
     detectors: [/^\x89PNG\x0D\x0A\x1A\x0A/],
     dec: async () => {
-      await pngEncDecInit()
+      await pngEncDecPromise
       return {
-        decode: (buffer: Buffer | Uint8Array): Buffer => {
+        decode: (buffer: Buffer | Uint8Array) => {
           const imageData = pngEncDec.decode(buffer)
-          pngEncDec.cleanup()
           return imageData
         },
-      } as any
+      }
     },
     enc: async () => {
-      await pngEncDecInit()
-      await oxipngInit()
+      await pngEncDecPromise
+      await oxipngPromise
       return {
         encode: (
-          buffer: Buffer | Uint8Array,
+          buffer: Uint8ClampedArray | ArrayBuffer,
           width: number,
           height: number,
-          opts: any
+          opts: { level: number }
         ) => {
           const simplePng = pngEncDec.encode(
             new Uint8Array(buffer),
             width,
             height
           )
-          const imageData = oxipng.optimise(simplePng, opts.level)
-          oxipng.cleanup()
```

Comment:
I believe we need to keep these cleanup calls for the memory leaks that @shuding fixed, unless if there's no more leaks with the emscripten upgrade. Could you weigh in @shuding?

Reply:
This is generally OK I think (3. in #23565). Removing it will not leak more memory, but it won't clean up the allocated memory for the finished task. When a new task arrives, the memory will still get released and new memory is allocated for the new task. So the behavior is basically:
- Current canary: new task (100MB↑), GC (100MB↓), new task (100MB↑), GC (100MB↓), ...
- With this PR: new task (100MB↑), new task (unchanged), new task (unchanged), ...

For long-running servers, it's definitely ideal to keep this logic though (or fix it in the upstream lib).

Reply:
When I applied the patch that set `wasm = null`, it caused null reference errors during `next build` at `wasm.__wbindgen_add_to_stack_pointer(-16)`.

https://github.com/vercel/next.js/runs/3750453542#step:7:1154

Oddly enough, the `next dev` tests passed 🤔 

Maybe I can try the lazy loading suggestion in the [other comment](https://github.com/vercel/next.js/pull/29506#discussion_r721763643) to see if that works 🕵️